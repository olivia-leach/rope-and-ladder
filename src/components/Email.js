import React, { Component, Fragment } from 'react';
import { MDCTextField } from '@material/textfield/index';

import './Email/Email.scss'

class Email extends Component {
    constructor(props) {
        super(props)
        this.state = {
            EMAIL: '',
            FNAME: '',
            LNAME: '',
        }
    }

    onTextChange = (e) => {
        const { value, name } = e.target
        this.setState({
            [name]: value,
            submitted: false,
        })
    }

    componentDidMount() {
        const textFields = document.querySelectorAll('.mdc-text-field');
        for (const textField of textFields) {
            MDCTextField.attachTo(textField);
        }
    }

    getFormData() {
        const data = {
            EMAIL: this.state.EMAIL,
            FNAME: this.state.FNAME,
            LNAME: this.state.LNAME
            // status: 'subscribed',
            // merge_fields: {
                // FNAME: this.state.name,
                // LNAME: this.state.name
            // }
        }
        data.formDataNameOrder = JSON.stringify(data);
        return data;
    }

    submitForm = (e) => {
        e.preventDefault()
        this.setState({ submitting: true })
        const data = this.getFormData();
    
        // let encoded = Object.keys(data).map(function(k) {
        //     return encodeURIComponent(k) + '=' + encodeURIComponent(data[k])
        // }).join('&')

        // const url = 'https://script.google.com/macros/s/AKfycbx0hWsJBWiSzDOIAzJldsRg0wdj7NwwtFO6vFNDS_cqbKAbvtQe/exec'
        // const listId = '9e67587f52'
        const url = 'https://ropeandladder.us4.list-manage.com/subscribe/post-json?u=929e4d360b1bc0291a85bc367&amp;id=51f155eaca&amp;c=?'
        const xhr = new XMLHttpRequest()
        xhr.open('GET', url)
        xhr.setRequestHeader('Content-Type', 'application/json; charset=utf-8')
        xhr.send(data)
        xhr.onreadystatechange = function() {
            console.log(xhr.responseText)
            if (xhr.status === 200) {
            this.setState({
                submitted: true,
                submitting: false,
            })
            }
            return;
        }.bind(this);
    
        // url encode form data for sending as post data
    }


    render() {
        const emailValid = this.state.EMAIL.includes('@') && this.state.EMAIL.includes('.')
        const canSubmit = !!this.state.FNAME && !!this.state.LNAME && emailValid
        return (
        <div className='email-container'>
            {this.state.submitted || this.state.submitting ? <h1 className='call-to-action-instructions text-center'>Thanks. We'll be in touch soon.</h1> :
            <Fragment>
                <div className='sign-up-text'>
                    <p>SIGN UP TO RECEIVE ALBUM UPDATES</p>
                    <p>AND TO ACCESS THE FIRST FIVE SONGS ON THE RECORD</p>
                </div>
                <div id="mc_embed_signup">
                    <form action="https://ropeandladder.us4.list-manage.com/subscribe/post?u=929e4d360b1bc0291a85bc367&amp;id=51f155eaca" target="_blank" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" className="flex flex-center input-container" novalidate>
                        <div className="mdc-text-field">
                            <input type="text" value={this.state.FNAME} name="FNAME" className="mdc-text-field__input"  id="mce-FNAME" onChange={this.onTextChange} />
                            <label className="mdc-floating-label" htmlFor="mce-FNAME">First Name </label>
                            <div className="mdc-line-ripple" />
                        </div>
                        <div className="mdc-text-field">
                            <input type="text" value={this.state.LNAME} name="LNAME" className="mdc-text-field__input"  id="mce-LNAME" onChange={this.onTextChange} />
                            <label className="mdc-floating-label" htmlFor="mce-LNAME">Last Name </label>
                            <div className="mdc-line-ripple" />
                        </div>
                        <div className="mdc-text-field">
                            <input type="email" value={this.state.EMAIL} name="EMAIL" className="mdc-text-field__input" id="mce-EMAIL" onChange={this.onTextChange} />
                            <label className="mdc-floating-label" htmlFor="mce-EMAIL">Email Address </label>
                            <div className="mdc-line-ripple" />
                        </div>
                        <div id="mce-responses" className="clear">
                            <div className="response" id="mce-error-response" style={{display: 'none'}}></div>
                            <div className="response" id="mce-success-response" style={{display: 'none'}}></div>
                        </div>
                        <div style={{ position: 'absolute', left: '-5000px' }} aria-hidden="true"><input type="text" name="b_929e4d360b1bc0291a85bc367_51f155eaca" tabindex="-1" value="" /></div>
                        <input disabled={!canSubmit} type='submit' value={this.state.submitting ? 'Signing up...' : 'Subscribe'} name="subscribe" id="mc-embedded-subscribe" className="mdc-button mdc-button--raised subscribe" />
                    </form>
                </div>


                    {/*<div>
                        <div className="mdc-text-field">
                        <input type="text" name='name' id="name-field" className="mdc-text-field__input" onChange={this.onTextChange} />
                        <label className="mdc-floating-label" htmlFor="name-field">
                            NAME
                        </label>
                        <div className="mdc-line-ripple" />
                        </div>
                    </div>
                    <div>
                        <div className="mdc-text-field">
                        <input type="email" name='email' id="email-field" className="mdc-text-field__input" onChange={this.onTextChange} />
                        <label className="mdc-floating-label" htmlFor="email-field">
                            EMAIL
                        </label>
                        <div className="mdc-line-ripple" />
                        </div>
                    </div>
                    <button className="mdc-button mdc-button--raised subscribe" disabled={!canSubmit || this.state.submitting || this.state.submitted} onClick={this.submitForm}>
                        {this.state.submitting ?
                            <span className="mdc-button__label">Signing up...</span>
                        :
                        <span className="mdc-button__label">
                            {this.state.submitted ? 'Thank you' : 'Subscribe'}</span>}
                    </button>
                        </div>*/}
            </Fragment>}
            </div>
        )
    }
}

export default Email;
