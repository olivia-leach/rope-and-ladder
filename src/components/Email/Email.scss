$date-letter-spacing: 3px;
$break-small: 767px;

.sign-up-text {
    text-align: center;
    text-transform: uppercase;
    font-size: 0.85rem;
    letter-spacing: $date-letter-spacing;
    margin-left: $date-letter-spacing;
    font-weight: 400;
    color: rgba(255, 255, 255, 0.7);
    padding: 0 5rem;
    line-height: 2;
}

.coming-soon-text {
    text-align: center;
    text-transform: uppercase;
    font-size: 2rem;
    margin: 1rem 0;
    letter-spacing: $date-letter-spacing * 2;
    font-weight: 200;
    color: rgba(255, 255, 255, 0.7);

    @media screen and (max-width: $break-small) {
        font-size: 1.5rem;
        margin-bottom: 2rem;
    }
}

.email-container {
    margin-top: -50px;
    padding-top: 1rem;
    padding-bottom: 2rem;
    flex-grow: 1;
    position: relative;
    font-family: 'arial';
    width: 100%;
    
    .input-container {
        flex-wrap: wrap;
        div {
            max-width: 190px;

            &.clear {
                margin-left: 0;
            }
        }
    }
  }