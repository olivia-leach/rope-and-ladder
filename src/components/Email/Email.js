import React, { Component, Fragment } from 'react';
import { TextField, Button, ThemeProvider } from 'rmwc'

import '@material/textfield/dist/mdc.textfield.css';
import '@material/floating-label/dist/mdc.floating-label.css';
import '@material/notched-outline/dist/mdc.notched-outline.css';
import '@material/line-ripple/dist/mdc.line-ripple.css';
import '@material/ripple/dist/mdc.ripple.css';
import '@rmwc/icon/icon.css';

import './Email.scss'

class Email extends Component {
    constructor(props) {
        super(props)
        this.state = {
            EMAIL: '',
            FNAME: '',
            LNAME: '',
        }
    }

    onTextChange = (e) => {
        const { value, name } = e.target
        this.setState({
            [name]: value,
            submitted: false,
        })
    }

    getFormData() {
        const data = {
            EMAIL: this.state.EMAIL,
            FNAME: this.state.FNAME,
            LNAME: this.state.LNAME
        }
        data.formDataNameOrder = JSON.stringify(data);
        return data;
    }

    submitForm = (e) => {
      e.preventDefault()
      this.setState({ submitting: true })
      const data = this.getFormData();
      const url = 'https://ropeandladder.us4.list-manage.com/subscribe/post-json?u=929e4d360b1bc0291a85bc367&amp;id=51f155eaca&amp;c=?'
      const xhr = new XMLHttpRequest()
      xhr.open('GET', url)
      xhr.setRequestHeader('Content-Type', 'application/json; charset=utf-8')
      xhr.send(data)
      xhr.onreadystatechange = function() {
          console.log(xhr.responseText)
          if (xhr.status === 200) {
          this.setState({
              submitted: true,
              submitting: false,
          })
          }
          return;
      }.bind(this);
    }


    render() {
        const emailValid = this.state.EMAIL.includes('@') && this.state.EMAIL.includes('.')
        const canSubmit = !!this.state.FNAME && !!this.state.LNAME && emailValid && !this.state.b_929e4d360b1bc0291a85bc367_51f155eaca
        return (
        <div className='email-container dark-section'>
            {this.state.submitted || this.state.submitting ? <h1 className='call-to-action-instructions text-center'>Thanks. We'll be in touch soon.</h1> :
            <ThemeProvider options={{ primary: '#dd7d45', secondary: '#dd7d45' }}>
                <div className='sign-up-text'>
                  <p>SUBSCRIBE TO THE ROPE AND LADDER MAILING LIST</p>
                </div>
                <form action="https://ropeandladder.us4.list-manage.com/subscribe/post?u=929e4d360b1bc0291a85bc367&amp;id=51f155eaca" target="_blank" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" className="flex flex-center input-container" noValidate>
                  <TextField
                    label='First Name'
                    name='FNAME'
                    onChange={this.onTextChange}
                    value={this.state.FNAME}
                  />
                  <TextField
                    label='Last Name'
                    name='LNAME'
                    onChange={this.onTextChange}
                    value={this.state.LNAME}
                  />
                  <TextField
                    label='Email Address'
                    name='EMAIL'
                    onChange={this.onTextChange}
                    value={this.state.EMAIL}
                  />
                  <div style={{ position: 'absolute', left: '-5000px' }} aria-hidden="true">
                    <input type="text" name="b_929e4d360b1bc0291a85bc367_51f155eaca" tabIndex="-1" value={this.state.HONEYPOT} onChange={this.onTextChange} />
                  </div>
                  <Button disabled={!canSubmit} label={this.state.submitting ? 'Signing up...' : 'Subscribe'} raised type='submit' />
                </form>
              </ThemeProvider>}
            </div>
        )
    }
}

export default Email;
