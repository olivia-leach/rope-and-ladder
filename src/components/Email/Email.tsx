import React, { useCallback, useState } from 'react'
import { TextField, Button } from '@material-ui/core'

import './Email.scss'

const testProperties = {
  firstName: {
    'data-testid': 'firstNameInput'
  },
  lastName: {
    'data-testid': 'lastNameInput'
  },
  email: {
    'data-testid': 'emailInput'
  }
}

const honeyPotStyles = { position: 'absolute' as const, left: '-5000px' }

const Email = () => {
  const [email, setEmail] = useState('')
  const [firstName, setFirstName] = useState('')
  const [lastName, setLastName] = useState('')
  const [honeyPot, setHoneyPot] = useState('')

  const onEmailChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setEmail(e.target.value)
    },
    [],
  )

  const onFirstNameChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setFirstName(e.target.value)
    },
    [],
  )

  const onLastNameChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setLastName(e.target.value)
    },
    [],
  )

  const onHoneyPotChange = useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      setHoneyPot(e.target.value)
    },
    [],
  )

  const emailValid = email.includes('@') && email.includes('.')
  const canSubmit = !!firstName && !!lastName && emailValid && !honeyPot

  return (
    <div className="email-container dark-section">
      <div className="sign-up-text">
        <p>SUBSCRIBE TO THE ROPE AND LADDER MAILING LIST</p>
      </div>
      <form
        action="https://ropeandladder.us4.list-manage.com/subscribe/post?u=929e4d360b1bc0291a85bc367&amp;id=51f155eaca"
        className="flex flex-center input-container"
        data-testid="emailForm"
        target="_blank"
        method="post"
        id="mc-embedded-subscribe-form"
        name="mc-embedded-subscribe-form"
        noValidate
      >
        <TextField
          color="secondary"
          inputProps={testProperties.firstName}
          label="First Name"
          margin="dense"
          name="FNAME"
          onChange={onFirstNameChange}
          size="small"
          value={firstName}
          variant="filled"
        />
        <TextField
          color="secondary"
          inputProps={testProperties.lastName}
          label="Last Name"
          margin="dense"
          name="LNAME"
          onChange={onLastNameChange}
          size="small"
          value={lastName}
          variant="filled"
        />
        <TextField
          color="secondary"
          inputProps={testProperties.email}
          label="Email Address"
          margin="dense"
          name="EMAIL"
          onChange={onEmailChange}
          size="small"
          value={email}
          variant="filled"
        />
        <div
          style={honeyPotStyles}
          aria-hidden="true"
        >
          <input
            data-testid='honeyPot'
            type="text"
            name="b_929e4d360b1bc0291a85bc367_51f155eaca"
            tabIndex={-1}
            value={honeyPot}
            onChange={onHoneyPotChange}
          />
        </div>
        <Button
          color="secondary"
          data-testid='submitButton'
          disabled={!canSubmit}
          variant="contained"
          size="large"
          type="submit"
        >
          Subscribe
        </Button>
      </form>
    </div>
  )
}

export default Email
