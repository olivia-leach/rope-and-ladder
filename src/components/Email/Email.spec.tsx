import { render, fireEvent } from '@testing-library/react'
import React from 'react'
import Email from './Email'

describe('Email', () => {
  it('should match snapshot', () => {
    const { container } = render(<Email />)
    expect(container).toMatchSnapshot()
  })

  it('should enable submit button when all required info is supplied w/no honeypot', () => {
    const { getByTestId } = render(<Email />)

    expect(getByTestId('submitButton')).toHaveProperty('disabled', true)

    fireEvent.change(getByTestId('firstNameInput'), { target: { value: 'Bob' }})
    fireEvent.change(getByTestId('lastNameInput'), { target: { value: 'Bobbins' }})
    fireEvent.change(getByTestId('emailInput'), { target: { value: 'bob@bobbins.bob' }})
    expect(getByTestId('submitButton')).toHaveProperty('disabled', false)

    fireEvent.change(getByTestId('honeyPot'), { target: { value: 'delicious' }})
    expect(getByTestId('submitButton')).toHaveProperty('disabled', true)
  })
})
