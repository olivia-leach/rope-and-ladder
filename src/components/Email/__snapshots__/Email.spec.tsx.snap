// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email should match snapshot 1`] = `
<div>
  <div
    class="email-container dark-section"
  >
    <div
      class="sign-up-text"
    >
      <p>
        SUBSCRIBE TO THE ROPE AND LADDER MAILING LIST
      </p>
    </div>
    <form
      action="https://ropeandladder.us4.list-manage.com/subscribe/post?u=929e4d360b1bc0291a85bc367&id=51f155eaca"
      class="flex flex-center input-container"
      data-testid="emailForm"
      id="mc-embedded-subscribe-form"
      method="post"
      name="mc-embedded-subscribe-form"
      novalidate=""
      target="_blank"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
          data-shrink="false"
        >
          First Name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiFilledInput-colorSecondary MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
            data-testid="firstNameInput"
            name="FNAME"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
          data-shrink="false"
        >
          Last Name
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiFilledInput-colorSecondary MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
            data-testid="lastNameInput"
            name="LNAME"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense"
      >
        <label
          class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
          data-shrink="false"
        >
          Email Address
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorSecondary MuiFilledInput-colorSecondary MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
            data-testid="emailInput"
            name="EMAIL"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        style="position: absolute; left: -5000px;"
      >
        <input
          data-testid="honeyPot"
          name="b_929e4d360b1bc0291a85bc367_51f155eaca"
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge Mui-disabled Mui-disabled"
        data-testid="submitButton"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Subscribe
        </span>
      </button>
    </form>
  </div>
</div>
`;
